src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -99,63 +103,50 @@ class RestrictedPatientsService(
-    val dischargeDateTime = calculateDischargeDateTime(offenderNo, latestMovement.movementDate, latestMovement.movementTime)
+  private fun getExistingRestrictedPatientDischargeData(
+    latestMovement: MovementResponse,
+    offenderNo: String,
+  ): ExistingDischargeData {
+    val dischargeDateTime =
+      calculateDischargeDateTime(offenderNo, latestMovement.movementDate, latestMovement.movementTime)
     return ExistingDischargeData(
       dischargeTime = dischargeDateTime,
       fromLocationId = latestMovement.fromAgency!!,
       comment = latestMovement.commentText,
     )
   }
 
-  private fun calculateDischargeDateTime(offenderNo: String, movementDate: String?, movementTime: String?): LocalDateTime {
+  private fun calculateDischargeDateTime(
+    offenderNo: String,
+    movementDate: String?,
+    movementTime: String?,
+  ): LocalDateTime {
     try {
       return LocalDateTime.parse("${movementDate}T$movementTime")
     } catch (e: Exception) {
-      throw BadRequestException(errorCode = "DISCHARGE_TIME_INVALID", message = "Prisoner ($offenderNo) does not have a valid movement date/time")
-    }
-  }
-
-  private fun addRestrictedPatient(restrictedPatient: RestrictedPatient): RestrictedPatientDto {
-    val newRestrictedPatient = restrictedPatientsRepository.saveAndFlush(restrictedPatient)
-
-    dischargeOrRollBackAndThrow(newRestrictedPatient)
-
-    return transform(
-      newRestrictedPatient,
-      Agency(agencyId = restrictedPatient.fromLocationId),
-      Agency(agencyId = restrictedPatient.hospitalLocationCode),
-      Agency(agencyId = restrictedPatient.supportingPrisonId),
-    ).also {
-      domainEventPublisher.publishRestrictedPatientAdded(restrictedPatient.prisonerNumber)
-      telemetryClient.trackEvent(
-        "restricted-patient-added",
-        mapOf(
-          "prisonerNumber" to restrictedPatient.prisonerNumber,
-          "fromLocationId" to restrictedPatient.fromLocationId,
-          "hospitalLocationCode" to restrictedPatient.hospitalLocationCode,
-          "supportingPrisonId" to restrictedPatient.supportingPrisonId,
-          "dischargeTime" to restrictedPatient.dischargeTime.toString(),
-        ),
-        null,
+      throw BadRequestException(
+        errorCode = "DISCHARGE_TIME_INVALID",
+        message = "Prisoner ($offenderNo) does not have a valid movement date/time",
       )
     }
   }
 
-  private fun dischargeOrRollBackAndThrow(newRestrictedPatient: RestrictedPatient) {
-    try {
-      prisonApiGateway.dischargeToHospital(newRestrictedPatient)
-    } catch (e: Exception) {
-      // We have to operate out of a transaction because we want the prisoner to be added to the database immediately.
-      // Prisoner Search listens to any external movements so may call restricted patients before this service ends.
-      // We therefore have to delete the patient manually if the call to Prison API fails
-      restrictedPatientsRepository.delete(newRestrictedPatient)
```

Comment:
we therefore now don't need to do a manual delete if the request to Prison API fails, it will be handled by the transaction instead